homeassistant/components/group/sensor.py

```diff
@@ -352,60 +343,44 @@ def __init__(
         self._state_incorrect: set[str] = set()
         self._extra_state_attribute: dict[str, Any] = {}
 
-    async def async_added_to_hass(self) -> None:
-        """When added to hass."""
-        for entity_id in self._entity_ids:
-            if self.hass.states.get(entity_id) is None:
-                self.calculate_attributes_later = async_track_state_change_event(
-                    self.hass, self._entity_ids, self.calculate_state_attributes
-                )
-                break
-        if not self.calculate_attributes_later:
-            await self.calculate_state_attributes()
-        await super().async_added_to_hass()
-
-    async def calculate_state_attributes(
-        self, event: Event[EventStateChangedData] | None = None
-    ) -> None:
+    def calculate_state_attributes(self, valid_state_entities: list[str]) -> None:
         """Calculate state attributes."""
-        for entity_id in self._entity_ids:
-            if self.hass.states.get(entity_id) is None:
-                return
-        if self.calculate_attributes_later:
-            self.calculate_attributes_later()
-            self.calculate_attributes_later = None
-        self._attr_state_class = self._calculate_state_class(self._state_class)
-        self._attr_device_class = self._calculate_device_class(self._device_class)
+        self._attr_state_class = self._calculate_state_class(
+            self._state_class, valid_state_entities
```

Comment:
Maybe we should rename `_state_class` to `_fixed_state_class` or `_configured_state_class` to make its purpose clearer? Same comment for device class and uom.